crates/router/src/core/payouts/validator.rs

```diff
@@ -13,26 +14,24 @@ use crate::{
     utils,
 };
 
-#[cfg(feature = "payouts")]
 #[instrument(skip(db))]
 pub async fn validate_uniqueness_of_payout_id_against_merchant_id(
     db: &dyn StorageInterface,
     payout_id: &str,
     merchant_id: &str,
+    storage_scheme: storage::enums::MerchantStorageScheme,
 ) -> RouterResult<Option<storage::Payouts>> {
     let payout = db
-        .find_payout_by_merchant_id_payout_id(merchant_id, payout_id)
+        .find_payout_by_merchant_id_payout_id(merchant_id, payout_id, storage_scheme)
         .await;
     match payout {
         Err(err) => {
-            if err.current_context().is_db_not_found() {
-                // Empty vec should be returned by query in case of no results, this check exists just
-                // to be on the safer side. Fixed this, now vector is not returned but should check the flow in detail later.
-                Ok(None)
-            } else {
-                Err(err
+            let erc = err.current_context();
+            match erc {
+                StorageError::ValueNotFound(_) => Ok(None),
+                _ => Err(err
                     .change_context(errors::ApiErrorResponse::InternalServerError)
-                    .attach_printable("Failed while finding payout_attempt, database error"))
+                    .attach_printable("Failed while finding payout_attempt, database error")),
             }
         }
```

Comment:
I don't think this is the correct way of implementing this function, can be confusing for some instance

Reply:
Re-looking at this function, it looks good to me.

It checks whether or not the resource exists in DB, and returns an optional result

If Err is returned, it was due to external DB issues.
If Ok(None) is returned, DB call was successful but payout was not found
If Ok(Some) is returned, DB call was successful and payout with given ID already exists.

Reply:
you can have an function named something like `get_optional_payout_by_merchant_id_payout_id` in PayoutInterface which would be calling `generic_find_by_id_optional`